File: src/main/java/pack/web/param/PaymentRq.java
package pack.web.param;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.math.BigDecimal;
import java.util.List;

import static java.math.RoundingMode.HALF_UP;

public class PaymentRq implements Rq {
    @JsonProperty("request_id")
    private String requestId;
    @JsonProperty("target_acc_id")
    private String toId;
    private BigDecimal amount;

    public String getRequestId() {
        return requestId;
    }

    public void setRequestId(String requestId) {
        this.requestId = requestId;
    }

    public String getToId() {
        return toId;
    }

    public void setToId(String toId) {
        this.toId = toId;
    }

    public BigDecimal getAmount() {
        return amount;
    }

    public void setAmount(BigDecimal amount) {
        this.amount = amount.setScale(2, HALF_UP);
    }

    @Override
    public List<String> getLocks() {
        return List.of(toId);
    }
}

File: src/main/kotlin/pack/web/param/PaymentRq.kt
package pack.web.param

import com.fasterxml.jackson.annotation.JsonProperty
import pack.web.Validator.checkAndScale
import pack.web.Validator.requireNotBlank
import java.math.BigDecimal

class PaymentRq(@JsonProperty("request_id") requestId: String?,
                @JsonProperty("target_acc_id") toId: String?,
                @JsonProperty("amount") amount: BigDecimal?) : Rq {
    val requestId: String = requireNotBlank(requestId, "request_id")
    val toId: String = requireNotBlank(toId, "target_acc_id")
    val amount: BigDecimal = checkAndScale(amount)

    override val locks: List<String>
        get() = listOf(toId)
}
